src/geneweaver/api/controller/species.py

```diff
@@ -0,0 +1,24 @@
+"""Endpoints related to species."""
+from typing import Optional
+
+from fastapi import APIRouter, Depends, Query
+from geneweaver.api import dependencies as deps
+from geneweaver.api.services import species as species_service
+from geneweaver.core.enum import GeneIdentifier
+from typing_extensions import Annotated
+
+router = APIRouter(prefix="/species", tags=["species"])
+
+
+@router.get("/")
+def get_species(
+    cursor: Optional[deps.Cursor] = Depends(deps.cursor),
+    taxonomy_id: Annotated[
+        Optional[int], Query(format="int64", minimum=0, maxiumum=9223372036854775807)
+    ] = None,
+    gene_id_type: Optional[GeneIdentifier] = None,
+) -> dict:
+    """Get species."""
+    response = species_service.get_species(cursor, taxonomy_id, gene_id_type)
```

Comment:
Just a note: I don't love that the API will return an int for the reference_gene_id_type, but I think we can tackle that in another PR. The right solution would probably be to use pydantic for the return but I would want to validate the schemas before we add that in.

Reply:
Changed